specs/eip4844/p2p-interface.md

```diff
@@ -139,45 +155,56 @@ Per `context = compute_fork_digest(fork_version, genesis_validators_root)`:
 | `BELLATRIX_FORK_VERSION` | `bellatrix.SignedBeaconBlock` |
 | `CAPELLA_FORK_VERSION`   | `capella.SignedBeaconBlock`   |
 
-#### BeaconBlockAndBlobsSidecarByRoot v1
+No more than `MAX_REQUEST_BLOCKS_EIP4844` may be requested at a time.
 
-**Protocol ID:** `/eth2/beacon_chain/req/beacon_block_and_blobs_sidecar_by_root/1/`
+#### BlobSidecarsByRoot v1
+
+**Protocol ID:** `/eth2/beacon_chain/req/blob_sidecars_by_root/1/`
 
 Request Content:
 
+```python
+class BlobIdentifier(Container):
+    block_root: Root
+    index: uint64
+```
+
 ```
 (
-  List[Root, MAX_REQUEST_BLOCKS]
+  List[BlobIdentifier, MAX_REQUEST_BLOCKS_EIP4844]
```

Comment:
Re-using the same `MAX_REQUEST_BLOCKS_EIP4844` here seems really weird because these requests aren't 1 to 1 with blocks. Even if it's still set to 128 (in practice we wouldn't query fore more than the number of blobs per block) I don't think the constant should be re-used

@divagant-martian made the point this would bite us if we ever wanted blobs per block > `MAX_REQUEST_BLOCKS_EIP4844`

Reply:
good point - I've introduced a separate constant _and_ adjusted to limits so that the limits always allow downloading 128 blocks _and_ their corresponding blobs in one request - for by-root, this means allowing a larger request - with current constants, we end up allowing 512 blobs, which data-size-wise has the same worst case as a 128-entry by-range request.

wdyt?

ffc78e99282e65cc4da92c4dba0743be434d367b

Reply:
Yea that looks good!